acra/acra-in-depth/security-design/_index.md

```diff
@@ -2,3 +2,165 @@
+- Attempts to extract whole tables with sensitive data by "poisoning" them with records that would never be selected unless SELECT * is used.
+- Inaccurate connection attempts to AcraConnector from the processes that don't match the expected conditions (if the users running AcraConnector / App itself are on the same VM / physical machine). Such expected conditions can be process names, name substrings, etc.
+
+We plan to include more triggers into the future releases.
+
+Once the trigger is triggered, an automatic reaction can be executed. You can choose to either:
+- notify a 3rd party monitoring / alarm system or
+- stop data decryption.
+
+### Guarantee 3: Add encryption context
+
+By differentiating the sensitive data via [Zones]({{< ref "acra/acra-in-depth/cryptography-and-key-management/#zones-INVALID" >}}), Acra users can build an additional layer of security:
+- Zone keys are used for encrypting the AcraStruct's RK section. In this case, AcraServer will only be able to pick the right key and decrypt the data by knowing the correct zone. This limits the leak scope for every private key, even if AcraServer is partially compromised.
+- Zone IDs are used as context when encrypting Secure Cell, providing additional protection.
+
+Zone data is supplied to an app via JSON (i.e. `{"id": "...", "public_key": "..."}`) from [HTTP api]({{< ref "acra/configuring-maintaining/general-configuration/acra-server.md#http-api-INVALID">}}) or [console utility]({{< ref "acra/security-controls/transport-security/acra-connector.md#acraconnector-and-acrawriter-INVALID" >}}).
+
+
+## Security model for Acra
+
+### Possible threats
+
+> Note: We recommend taking a look at the [architectural scheme of Acra]({{< ref "acra/acra-in-depth/data-flow/#-INVALID" >}}) before continuing to read.
+
+Acra at its core is a set of tools that allow safeguarding the security of a database (running PostgreSQL or MySQL) against the known widespread threats.
+
+The most dangerous current known security threats are:
+- Threats related to excess / abuse of access privileges.
+- Data leaks caused by mistakes in deploy, configuration settings, backup thefts, etc.
+- SQL injections.
+- Denial of service.
+- Vulnerabilities in the database protocol.
+- Weak audit or an absence thereof.
+- Operational system vulnerabilities.
+- Unsafe handling of cryptographic keys.
+
+### Conditions of secure work
+Acra can perform its protective functions properly and protect from the security threats 1, 2, 3 if the following security assumptions are met:
+- The PKI infrastructure is trusted;
+- AcraServer is trusted;
+- The client is less trusted than the server.
+
+### Possible consequences of compromisation
+Let’s consider all the possible consequences of any of separate component being broken (broken as in “fully compromised” when the adversary fully overtakes the work of the component and gains full access to its memory).
```

Comment:
> Possible consequences of **compromisation**

I don't think “compromisation” is a widely used word. “Data breach” or “security breach” would be something more common and easier to grasp.

> **all** the possible consequences

Does this cover zombie infestation and the DBAs losing their jobs? If no, I guess it should be just “possible consequences”.

Reply:
> I don't think “compromisation” is a widely used word. “Data breach” or “security breach” would be something more common and easier to grasp.

I tried to google it and IMHO, it's a common word in data/app security context, used in news, articles, etc.
https://www.securityweek.com/search/google/compromise?query=compromise&cx=016540353864684098383%3A6mcx-eenlzi&cof=FORID%3A11&sitesearch=&safe=off
@storojs72 , what do you think? IMHO "data breach" is more often used in general hype news when something was hacked without knowing what was done. But compromisation is a more specific action.

> Does this cover zombie infestation and the DBAs losing their jobs? If no, I guess it should be just “possible consequences”.

there we describe what we mean in parentheses. I don't know why you think about zombies in docs for data security tool ) and why someone will think about same too.

Reply:
"compromisation" – this is a very rarely used word in security context. Say breach, incident, compromise, not compromisation.

Reply:
I would agree that term "data breach" or "hacking" is usually used in non-formal sources (hype news as well), although at recent time "breach" term starts appearing in applied RnD papers like [this](https://arxiv.org/pdf/1703.02014.pdf) where they use "system breach" and "data breach" as similar meanings.

In more formal sources (especially crypto-related) I usually meet "adversarial breaking" term (for example [here](https://www.cs.virginia.edu/dwu4/courses/sp20/static/reductions.pdf)). It denotes to situation when the security of system / cryptographic scheme is formalized in a form of some mathematical problem and adversary under some circumstances is able to "break security".

Having that, I would not mind against "breach", but IMO I would use "Analyzing of possible consequences of Acra breaking" for naming of this section